nutils/sample.py

```diff
@@ -245,13 +245,13 @@ def __call__(self, __func: function.IntoArray) -> function.Array:
         indices = evaluable.loop_concatenate(evaluable._flat(self.get_evaluable_indices(ielem)), ielem)
         return _ReorderPoints(_ConcatenatePoints(func, self), indices) * funcscale
 
-    def basis(self) -> function.Array:
+    def basis(self, interpolation: str = 'none') -> function.Array:
         '''Basis-like function that for every point in the sample evaluates to the
         unit vector corresponding to its index.'''
 
         raise NotImplementedError
 
-    def asfunction(self, array: numpy.ndarray) -> function.Array:
+    def asfunction(self, array: numpy.ndarray, interpolation: str = 'none') -> function.Array:
```

Comment:
Can you document the `interpolation` argument?

Reply:
added "Interpolation scheme used to map sample values to the evaluating sample. Valid values are "none", demanding that the evaluating sample mathes self, or "nearest" for nearest-neghbour mapping."

Reply:
Typo: 'neghbour' should be 'neighbour'.

Reply:
fixed